lib/operator/subtract.js

```diff
@@ -102,7 +94,9 @@ class SubtractSubstractedObserver { // implements CollectionObserver
   }
 
   added(items) {
-    this.resultColl.removeAll(items);
+    for (let item of items) {
+      this.resultColl.removeEach(item);
+    }
```

Comment:
heh! Here, the old code had the bug described in `removed()`, and your new code fixes it. Good change! :-)